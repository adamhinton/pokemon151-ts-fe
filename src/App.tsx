import React, { ReactElement } from "react";
import "./App.css";
import AllPokemon from "./components/AllPokemon";
import Header from "./components/Header";
import pokemonData from "./data";
import GlobalStyle from "./globalStyles";

const App = (): ReactElement => {
  return (
    <div className="App">
      <GlobalStyle />
      <Header />
      <h1>Adam Hinton</h1>
      <AllPokemon isShiny={true} pokemonData={pokemonData} />
    </div>
  );
};

export default App;
