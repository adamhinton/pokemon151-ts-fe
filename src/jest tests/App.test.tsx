import { render, screen } from "@testing-library/react";
import App from "../App";
import renderer from "react-test-renderer";

jest.mock("../components/AllPokemon", () => () => {
  return <div>This is a mocked AllPokemon Component</div>;
});

test("[1] Renders without errors", () => {
  render(<App />);
});

test("[2] Renders child components", () => {
  render(<App />);

  const mockedAllPokemonComponent = screen.getByText(
    "This is a mocked AllPokemon Component"
  );
  expect(mockedAllPokemonComponent).toBeVisible();
});

test("[3] Matches snapshot from 9.18.22", () => {
  const component = renderer.create(<App />);

  let tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});
