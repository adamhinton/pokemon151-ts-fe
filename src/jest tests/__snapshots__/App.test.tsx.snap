// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[3] Matches snapshot from 9.18.22 1`] = `
<div
  className="App"
>
  <header>
    <h1
      className="header-item header-h1"
    >
      Pokemon 151 Project
    </h1>
    <section
      className="darkmode-shiny-dropdown"
    >
      <menu
        className="darkmode-shiny-section"
      >
        <div
          className="darkmode-section"
        >
          <div
            className="logo"
            data-testid="logo"
          />
        </div>
        <div
          className="vertical-line"
        />
      </menu>
    </section>
    <section
      className="adam-info header-item"
    >
      <h2>
        Author: Adam Hinton
      </h2>
    </section>
  </header>
  <h1>
    Adam Hinton
  </h1>
  <div>
    This is a mocked AllPokemon Component
  </div>
</div>
`;
